projects/storefrontlib/src/events/product/product-page-event.builder.ts

```diff
@@ -72,23 +72,20 @@ export class ProductPageEventBuilder {
       skip(1)
     );
 
-    const categoryPageEvent$ = this.eventService.get(PageEvent).pipe(
-      map((pageEvent) => ({
-        isCategoryPage: pageEvent.semanticRoute === 'category',
-        categoryCode: pageEvent.context.id,
-      }))
-    );
-
-    return categoryPageEvent$.pipe(
+    return this.eventService.get(PageEvent).pipe(
       switchMap((pageEvent) => {
-        if (!pageEvent.isCategoryPage) {
+        if (pageEvent?.semanticRoute !== 'category') {
           return EMPTY;
         }
 
         return searchResults$.pipe(
           map((searchResults) => ({
-            categoryCode: pageEvent.categoryCode,
-            categoryName: searchResults.breadcrumbs[0].facetValueName,
+            ...pageEvent,
+            categoryCode: pageEvent?.context?.id,
```

Comment:
I means to add the `categoryCode` and `numberOfResults` to be conditionally added, similar to the `categoryName`

Reply:
Addressed above